### ISSUE TITLE
Creating Container-Based Apps - URL for the API project

### ISSUE THREAD

**Issue by eliassal** [bmaluijb/BuildingCloudNativeSolutions#3]
Hi Barry, In home controller for the branch (withContainers), You use the following

`request.RequestUri = new Uri("http://nationalcookies.api/api/values/");`
whereas in the properties page of the API project it runs on 
`http://localhost:61157`

how it can magically, the mvc communicate with this url (no DNS entrey, not host file........) run as you indicated? is it a url that needs to be configured somewhere? when I hit F5, I have the normal mvc that runs without creating a container.
Can you please tell me where I can find the config for docker support to double check

Here is the error I am getting

![ContainerErr](https://user-images.githubusercontent.com/8447864/87702484-c22eeb80-c799-11ea-8ab3-cbd42f1de3a9.jpg)

**Comment by bmaluijb** [comment]
Hi, you can check the docker files in the solution. Those configure the routing between the applications (api and website). Did you follow the module in the course and have all the prerequisites installed to run containers?

**Comment by eliassal** [comment]
Yes, all azure workloads and docker desktop last version

**Comment by eliassal** [comment]
I went through docker files, not able to see where the url is configured

**Comment by eliassal** [comment]
Barry, the container branch is not compiling correctly. There is something strange regarding the MVC project. When I right click the project, I see only 

> Docker support

but no
`Orchestrator` container support 
as you can notice in the snapshot 1 but doing the same thing in the API project I have this option

Wen I try to compile, it requires me to connect to Azure and not to compile locally

![ContainerOrchSupport_003](https://user-images.githubusercontent.com/8447864/87803490-d4209500-c852-11ea-9fe0-60688678b26f.jpg)

**Snapshot 1**

![ContainerOrchSupport_002](https://user-images.githubusercontent.com/8447864/87803574-f4e8ea80-c852-11ea-9d9f-9d696719e672.jpg)

**Snapshot** Api project**
![ContainerOrchSupport](https://user-images.githubusercontent.com/8447864/87803677-219d0200-c853-11ea-87a8-21fdcca73abe.jpg)

**Comment by eliassal** [comment]
Barry, can you please get back with instructions how to setup the code for running the version with Containers, I am completely blocked. Whe I hit F5 I am always presented with my email used for my Azure  account but no subscription  appears
What should be the code in appsettings.json in both the mvc and api in this version? Thanks
![ContainerOrchSupport_004](https://user-images.githubusercontent.com/8447864/87957689-e0099280-cab0-11ea-9254-955e05a4eada.jpg)

**Comment by bmaluijb** [comment]
Have you finished the course on Pluralsight? The code works out of the box for me, so I’m not sure what is going wrong in your end

**Comment by eliassal** [comment]
Yes, I finished the course 2 months back, so I am using the pdf files and notes I took whicle watching the videos. Tell me for one thing or share the config of your MVC container appsettings.json please, how you configure the container to connect to  localhost:8081, so it will then think that it has the cosmosdb inside the container whereas it is on host localhost:8081, no?
Once I get this, I will try to rebuilt everything for the containers version

**Comment by eliassal** [comment]
OK Barry, I have rewritten the whole API project and now I am getting composer running and creating the 2 containers, however, as far as I am aware, it not possible to put 

`  "CosmosDBEndpoint": "https://localhost:8081/"`

in tha ppsettings, as when the container runs it will look for this on its web site not the host where cosmosdb is installed, can you please provide the settings you use?

**Comment by eliassal** [comment]
Barry, while waiting for you to respond, I managed to make the solution works after  i did a couple of changes, I started cosmosdb emulator with networks support 
`Microsoft.Azure.Cosmos.Emulator.exe /AllowNetworkAccess`
then  changed  the appsettings in the mvc  to the following
`"CosmosDBEndpoint": "https://192.168.10.3:8081/",`

However, I am not able to understand how all the plumbing is working. 
Issuing 
`docker ps `
shows
CONTAINER ID        IMAGE                     COMMAND               CREATED             STATUS              PORTS                                           NAMES
d8f28db8c573        nationalcookiesmvc:dev    "tail -f /dev/null"   3 hours ago         Up 3 hours          0.0.0.0:32773->80/tcp, 0.0.0.0:32772->443/tcp   NationalCookies.MVC
3d882282b089        nationalcookiessapi:dev   "tail -f /dev/null"   3 hours ago         Up 3 hours          0.0.0.0:32771->80/tcp, 0.0.0.0:32770->443/tcp   NationalCookiesS.API

NationalCookiesS.API exposes port 80 through port 32771, I am not able to understand how access can be done through port 80 in the home controller using 
`request.RequestUri = new Uri("http://nationalcookiesS.api/api/values/");`

Can you please explain?


"CosmosDBEndpoint": "https://192.168.1.3:8081/",

**Comment by eliassal** [comment]
Barry, can't understand why you are not able to provide an explanation, thanks anyway
